src/huggingface_hub/inference/_mcp/mcp_client.py

```diff
@@ -0,0 +1,236 @@
+import json
+import logging
+import warnings
+from contextlib import AsyncExitStack
+from pathlib import Path
+from typing import TYPE_CHECKING, AsyncIterable, Dict, List, Optional, Union
+
+from typing_extensions import TypeAlias
+
+from ...utils._runtime import get_hf_hub_version
+from .._generated._async_client import AsyncInferenceClient
+from .._generated.types import (
+    ChatCompletionInputMessage,
+    ChatCompletionInputTool,
+    ChatCompletionStreamOutput,
+    ChatCompletionStreamOutputDeltaToolCall,
+)
+from .._providers import PROVIDER_OR_POLICY_T
+from .utils import format_result
+
+
+if TYPE_CHECKING:
+    from mcp import ClientSession
+
+logger = logging.getLogger(__name__)
+
+# Type alias for tool names
+ToolName: TypeAlias = str
+
+
+class MCPClient:
+    def __init__(
+        self,
+        *,
+        model: str,
+        provider: Optional[PROVIDER_OR_POLICY_T] = None,
+        api_key: Optional[str] = None,
+    ):
+        warnings.warn(
+            "'MCPClient' is experimental and might be subject to breaking changes in the future without prior notice.",
+            UserWarning,
+        )
```

Comment:
personally think we could omit this, but no strong opinion

Reply:
moved it the docstring